commands/azureCommands/delete-image.ts

```diff
@@ -14,34 +14,64 @@ import { Repository } from "../../utils/Azure/models/repository";
 import { getLoginServer } from "../../utils/nonNull";
 import * as quickPicks from '../utils/quick-pick-azure';
 
+/** Function to untag an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ */
+export async function untagAzureImage(context?: AzureImageTagNode): Promise<void> {
+    await removeImage(context, true);
+}
+
 /** Function to delete an Azure hosted image
  * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
  */
 export async function deleteAzureImage(context?: AzureImageTagNode): Promise<void> {
+    await removeImage(context, false);
+}
+
+/** Function to delete an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ * @param untag : if true deletes the image tag, otherwise removes digest and all other tags associeted to the image selected.
+ */
+async function removeImage(context: AzureImageTagNode, untag: boolean): Promise<void> {
+    let action: string = (untag) ? "untag" : "delete";
     let registry: Registry;
-    let repoName: string;
-    let tag: string;
+    let repo: Repository;
+    let image: AzureImage;
 
     if (!context) {
         registry = await quickPicks.quickPickACRRegistry();
-        const repository: Repository = await quickPicks.quickPickACRRepository(registry, 'Select the repository of the image you want to delete');
-        repoName = repository.name;
-        const image: AzureImage = await quickPicks.quickPickACRImage(repository, 'Select the image you want to delete');
-        tag = image.tag;
+        repo = await quickPicks.quickPickACRRepository(registry, `Select the repository of the image you want to ${action}`);
```

Comment:
I don't think these two functions share enough code to warrant combining into a single method. I think they would be better served as separate functions.

Reply:
I can separate them again, no problem.
Altho the main difference it's the info displayed to the user. Functionally speaking the one change is the content in the path variable.